pkg/hintrunner/zero/zerohint_keccak.go

```diff
@@ -102,3 +105,93 @@ func createKeccakWriteArgsHinter(resolver hintReferenceResolver) (hinter.Hinter,
 
 	return newKeccakWriteArgsHint(inputs, low, high), nil
 }
+
+func newBlockPermutationHint(keccakStateSizeFelts, keccakPtr hinter.ResOperander) hinter.Hinter {
+	name := "BlockPermutation"
+	return &GenericZeroHinter{
+		Name: name,
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			// from starkware.cairo.common.keccak_utils.keccak_utils import keccak_func
+			// _keccak_state_size_felts = int(ids.KECCAK_STATE_SIZE_FELTS)
+			// assert 0 <= _keccak_state_size_felts < 100
+
+			// output_values = keccak_func(memory.get_range(
+			// 	ids.keccak_ptr - _keccak_state_size_felts, _keccak_state_size_felts))
+			// segments.write_arg(ids.keccak_ptr, output_values)
+
+			keccakStateSize, err := hinter.ResolveAsUint64(vm, keccakStateSizeFelts)
+			if err != nil {
+				return err
+			}
+
+			keccakWritePtr, err := hinter.ResolveAsAddress(vm, keccakPtr)
+			if err != nil {
+				return err
+			}
+
+			// why does the need contains this check? keccakStateSize is a constant: 25
```

Comment:
Maybe it may be overwritten sometime? Or someone can change that, but that doesn't seem likely. I thing although it seems to be intuitive not to check that, maybe there is some edgecase that we don't see.

Reply:
yeah i'm trying to think of a reason. But the python VM code is clear, and declares a `KECCAK_STATE_SIZE_FELTS`  const. I dont see any reason for this valid to be something else than 25

Reply:
Yea makes sense, but still leave the check here, lets implement what is directly in the hint code